libs/error-stack/src/report.rs

```diff
@@ -243,7 +243,7 @@ use crate::{
 /// ```
 #[must_use]
 #[expect(clippy::field_scoped_visibility_modifiers)]
-pub struct Report<C: ?Sized> {
+pub struct Report<C: ?Sized = dyn Error + Send + Sync + 'static> {
```

Comment:
We don’t want to encourage people to use `dyn Error` as the common case. People should explicitly specify, if a `Report` is untyped. 
```suggestion
pub struct Report<C: ?Sized>
```
Also, we should add a documentation in which scenarios `Report<dyn Error>` is discouraged and when it’s useful:
- in return statements in an `Err` variant is discouraged, as a typed error is more representative. `Report<dyn Error>` does not add any contextual information to the `Report` which is a core principal of `error-stack`
- in argument position (e.g. for error handling) might be useful if the handling code does not need to know the current context. For example if it just searches for a `Suggestion` in the `Report` it doesn’t need to know `C`

The last bullet brings me to another point: We might need to make sure, that `current_context` is not callable on `Report<dyn Error>`. I haven’t looked at this, yet, but it might cause issues. Maybe it also just works.

Reply:
Also, the changelog needs updating

Reply:
> Yes, it was allowed but it was not possible to construct it (unless you use unsafe transmutation). I'd like to have some documentation around dyn Error before merging this.

Got it, cool! I've added a changelog entry, do you want to take a stab at documentation? I think the you have a much better feel for how you'd like it presented.